### PR TITLE
[13.0][FIX] s_p_group_by_partner_by_carrier: fix printing of delivery slip

### PR DESCRIPTION
**Pull request by sebalix**
Two bugfixes:
- filter out canceled moves
- in case there is no sale order related to some moves (even if this should
  not happen), do not print an empty section line

Ref. 2967

### CODE REVIEW

**Review by simahawk — APPROVED** [review]
seems legit

File: stock_picking_group_by_partner_by_carrier/models/stock_picking.py [hunk 1]
```diff
@@ -179,11 +180,12 @@ def get_delivery_report_lines(self):
...
                 for sale, sale_moves in grouped_moves:
-                    line_desc = sale.get_name_for_delivery_line()
-                    fake_record.update(
-                        {"description_picking": line_desc, "origin": sale.name}
-                    )
-                    sales_and_moves |= sales_and_moves.new(fake_record.copy())
+                    if sale:
+                        line_desc = sale.get_name_for_delivery_line()
+                        fake_record.update(
+                            {"description_picking": line_desc, "origin": sale.name}
```
Comment (simahawk) [inline stock_picking_group_by_partner_by_carrier/models/stock_picking.py:186]: hmm we'll miss the keys? Is there a check or is going to fail in the report?

**Review by mmequignon — APPROVED** [review]

**Review by TDu — APPROVED** [review]

File: stock_picking_group_by_partner_by_carrier/models/stock_picking.py (hunk 1 above)
Comment (sebalix) [reply stock_picking_group_by_partner_by_carrier/models/stock_picking.py:186]: This fake record dict is only used to append a section line with the name of the sale order, if there is no order we don't want to create such line.

**Pull request merged**